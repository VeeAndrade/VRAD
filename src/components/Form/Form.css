.user-input-form {
  height: 400px;
  width: 550px;
  display: flex;
  background: linear-gradient(rgba(189, 189, 189, 0.63), rgba(255, 255, 255, 0.749));
  align-self: center;
  align-items: center;
  justify-content: center;
  flex-flow: column nowrap;
  margin-top: 9%;
  border-radius: 15px;
}

.username-input,
.useremail-input {
  width: 70%;
  height: 8%;
  margin-top: 5%;
  font-size: 1.1em;
  padding-left: 10px;
  border-radius: 5px;
  border: none;
}

.purpose-section {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  width: 70%;
  height: 10%;
  margin-top: 5%;
}

.purpose-label {
  display: flex;
  justify-content: center;
  align-items: center;
  color: #003459;
}

.purpose-selection {
  display: flex;
  width: 70%;
  height: 90%;
}

.continue-btn {
  display: flex;
  color: #fff;
  text-decoration: none;
  align-items: center;
  justify-content: center;
  margin-top: 8%;
  height: 10%;
  width: 72%;
  border-radius: 5px;
  background-color: #003459;
  font-size: 1.3em;
}

.continue-btn:hover {
  background-color: #007ea7;
  cursor: pointer;
}