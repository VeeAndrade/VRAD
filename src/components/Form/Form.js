import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types'
import './Form.css'


export default class Form extends Component {
  constructor() {
    super();
    this.state = {
      user: '',
      purpose: ''
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value })
  }
   updateUserInfo = (user, purpose) => {
    this.setState({ user })
    this.setState({ purpose })
  }

  validateUser = e => {
    if(!this.state.user || !this.state.purpose || !this.state.email) {
      window.alert('You must fill out all fields')
    } else {
      this.props.updateUserInfo(this.state.user, this.state.purpose);
    }
  }

  render() {
    return (
      <form className='user-input-form'>
        <input
          className='username-input'
          type='text'
          placeholder='Name'
          value={this.state.title}
          name='user'
          onChange={this.handleChange}
        />
        <input
          className = 'useremail-input'
          type='email'
          placeholder='Email'
          name='email'
          onChange={this.handleChange}
          required
        />
        <section className='purpose-section'>
        <label className='purpose-label'>Reason for trip: </label>
        <select name='purpose' className='purpose-selection' onChange={this.handleChange}>
          <option value="">Select one--</option>
          <option value='Vacation'>Vacation</option>
          <option value='Business'>Business</option>
          <option value='Other'>Other</option>
        </select>
        </section>
        <Link className='continue-btn' onClick={this.validateUser} to={() => {
          if (!this.state.user || !this.state.purpose || !this.state.email) {
            return '/'
          } else {
            return '/areas'
          }
        }}>
          Continue
        </Link>
      </form>
    )
  }
}

Component.propTypes = {
  user: PropTypes.string,
  purpose: PropTypes.string
}