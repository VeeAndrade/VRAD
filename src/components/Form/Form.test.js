import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import { shallow } from 'enzyme'

describe('Form', () => {
  let wrapper;
  const mockUpdateUserInfo = jest.fn();
  
  beforeEach(() => {
    wrapper = shallow(<Form updateUserInfo={mockUpdateUserInfo} />)
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should start with a default state', () => {
    expect(wrapper.state('user')).toEqual('')
    expect(wrapper.state('purpose')).toEqual('')
  });
  
  it('should update the state when handleChange is called', () => {
    const expectedName = 'Johnny'
    const expectedPurpose = 'Vacation'
    const mockEvent1 = {
      target: {
        name: 'user',
        value: 'Johnny'
      }
    }

    const mockEvent2 = {
      target: {
        name: 'purpose',
        value: 'Vacation'
      }
    }
    wrapper.instance().handleChange(mockEvent1)
    expect(wrapper.state('user')).toEqual(expectedName)
    
    wrapper.instance().handleChange(mockEvent2)
    expect(wrapper.state('purpose')).toEqual(expectedPurpose)
  });

  it('should be able to set the state', () => {
    wrapper.instance().updateUserInfo('Vee', 'Vacation')
    expect(wrapper.state('user')).toEqual('Vee')
    expect(wrapper.state('purpose')).toEqual('Vacation')
  });
})