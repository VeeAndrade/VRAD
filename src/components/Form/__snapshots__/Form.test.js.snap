// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="user-input-form"
>
  <input
    className="username-input"
    name="user"
    onChange={[Function]}
    placeholder="Name"
    type="text"
  />
  <input
    className="useremail-input"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
  />
  <section
    className="purpose-section"
  >
    <label
      className="purpose-label"
    >
      Reason for trip: 
    </label>
    <select
      className="purpose-selection"
      name="purpose"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Select one--
      </option>
      <option
        value="Vacation"
      >
        Vacation
      </option>
      <option
        value="Business"
      >
        Business
      </option>
      <option
        value="Other"
      >
        Other
      </option>
    </select>
  </section>
  <Link
    className="continue-btn"
    onClick={[Function]}
    to="/areas"
  >
    Continue
  </Link>
</form>
`;
